src/sentry/models/organization.py

```diff
@@ -160,10 +160,12 @@ def get_default(cls):
         Return the organization used in single organization mode.
         """
 
-        return cls.objects.get(
-            id=settings.SENTRY_ORGANIZATION,
+        if settings.SENTRY_ORGANIZATION is not None:
+            return cls.objects.get(id=settings.SENTRY_ORGANIZATION)
+
+        return cls.objects.filter(
```

Comment:
This is just bringing back the behavior seen before #13964, but only does a `SELECT` with an id when explicitly setting one.